apps/www/_blog/2023-11-01-supabase-is-now-compatible-with-nextjs-14.mdx

```diff
@@ -0,0 +1,192 @@
+  const { data } = await supabase.from('...').select()
+  return ...
+}
+```
+
+Now with Server Actions, mutation logic can be co-located with the Server Components that fetch data and render the page:
+
+```tsx
+export default async function Page() {
+  const { data } = await supabase.from('...').select()
+
+  const createNote = async () => {
+    'use server'
+    await supabase.from('...').insert({...})
+  }
+
+  return ...
+}
+```
+
+> To hear more about our thoughts on Next.js Conf and the release of Next.js 14, check out our [Twitter space](https://twitter.com/i/spaces/1yoKMwNWbRjJQ?s=20). [Yuri](https://twitter.com/yuricodesbot), [Alaister](https://twitter.com/alaisteryoung), [Terry](https://twitter.com/saltcod) and [myself](https://twitter.com/jonmeyers_io) talk through how we use Next.js at Supabase and what we personally found most exciting about Next.js Conf and the release of Next.js 14.
+
+## Is Supabase compatible with Next.js 14?
+
+Yes, it is! So much so that [Guillermo Rauch](https://twitter.com/rauchg) shouted us out in the keynote!
+
+<img
+  className="dark:hidden !m-0"
+  alt="Tweet from Guillermo Rauch showing Supabase as one of the companies compatible with the Next.js App Router - light mode version"
+  src="/images/blog/2023-11-01-supabase-is-now-compatible-with-nextjs-14/guillermo-tweet-light.png"
+/>
+
+<img
+  className="hidden dark:block !m-0"
+  alt="Tweet from Guillermo Rauch showing Supabase as one of the companies compatible with the Next.js App Router - dark mode version"
+  src="/images/blog/2023-11-01-supabase-is-now-compatible-with-nextjs-14/guillermo-tweet-dark.png"
+/>
+
+Since the release of the [App Router](https://nextjs.org/docs/app)—launched as beta with Next.js 13—we have been working closely with the team at Vercel to ensure that Supabase is fully compatible with every part of Next.js.
+
+So for the [App Router](https://nextjs.org/docs/app), that's:
+
+- [Client Components](https://nextjs.org/docs/app/building-your-application/rendering/client-components)
+- [Server Components](https://nextjs.org/docs/app/building-your-application/rendering/server-components)
+- [Route Handlers](https://nextjs.org/docs/app/building-your-application/routing/route-handlers)
+- [Server Actions](https://nextjs.org/docs/app/api-reference/functions/server-actions)
+- [Middleware](https://nextjs.org/docs/app/building-your-application/routing/middleware)
+
+And for the [Pages Router](https://nextjs.org/docs/pages):
+
+- [`getServerSideProps` function](https://nextjs.org/docs/pages/building-your-application/data-fetching/get-server-side-props)
+- [`getStaticProps` function](https://nextjs.org/docs/pages/building-your-application/data-fetching/get-static-props)
+- [API Routes](https://nextjs.org/docs/pages/building-your-application/routing/api-routes)
+- [Page Components](https://nextjs.org/docs/pages/building-your-application/data-fetching/client-side)
+
+## What problem needed to be solved?
```

Comment:
This heading caught me a bit out of the blue. What problem needed to be solved with what? With Supabase? Next? Auth helpers? A new heading and maybe a transition sentence might be useful

Reply:
Good call! Went with a transition sentence:

So why does it require work on the Supabase side to make it compatible with Next.js?

And then new heading: Configuring Supabase to use Cookies